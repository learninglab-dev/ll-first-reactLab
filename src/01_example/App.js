import React from 'react'
import Layout from './Layout'
import Switch from './Switch'
import imgSources from './data'


export default function App() {
  const imgToGifs = imgSources.map((urls, i) => {
    return <Switch img={urls.img} gif={urls.gif} alt={i} key={i} />
  })
  return (
    <Layout>
      {imgToGifs}
    </Layout>
  )
}
