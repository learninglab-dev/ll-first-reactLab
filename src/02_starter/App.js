import React from 'react'
import Layout from './Layout'
import imgSources from './data'
import Image from './Img'
import Switch from './Switch'
import { Row, Col } from 'shards-react'


//the top-level component for our app. I've added some basic styling and a flexbox layout for you so your gifs make a nice grid :)
export default function App() {
  //call Switch here rather than Image
  return (
    <Layout>
      <Switch
        imgUrl='https://images.pexels.com/photos/416160/pexels-photo-416160.jpeg'
        gifUrl='https://media.giphy.com/media/HZUHce5TyYixq/giphy.gif'/>
      <Switch
        imgUrl='https://images.pexels.com/photos/127027/pexels-photo-127027.jpeg'
        gifUrl='https://media.giphy.com/media/VS3f7UiKom7hS/giphy.gif' />
     <Switch
        imgUrl='https://images.pexels.com/photos/1564506/pexels-photo-1564506.jpeg'
        gifUrl='https://media.giphy.com/media/XHY2TRKxJQfsrwS33v/giphy.gif' />
    </Layout>
  )
}
