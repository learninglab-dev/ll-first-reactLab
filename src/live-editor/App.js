import React, { useState } from 'react'
// import './App.css'
// import Playground from 'component-playground'
// import Button from './Button'
import {
  LiveProvider,
  LiveError,
  LivePreview
} from 'react-live'
import LiveEditor from './reactLive/LiveEditor'
import styled, { css } from 'styled-components'
import * as polished from 'polished'
import {reactLiveHome} from './themes/theme.js'

const background = '#42374a';
const foreground = '#f8f8f2';
const red = '#ff5555';
const blue = polished.lighten(0.2, '#98E342');
const lightGrey = polished.darken(0.06, '#272727');


const StyledProvider = styled(LiveProvider)`
  border-radius: ${polished.rem(3)};
  box-shadow: 1px 1px 20px rgba(20, 20, 20, 0.27);
  overflow: hidden;
  margin-bottom: ${polished.rem(100)};
`

const LiveWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: stretch;
  align-items: stretch;
  @media (max-width: window.innerWidth) {
    flex-direction: column;
  }
`;

const column = css`
  flex-basis: 50%;
  width: 100%;
  max-width: 100%;
  @media (max-width: 1000px) {
    flex-basis: auto;
    width: 100%;
    max-width: 100%;
  }
`;

const StyledEditor = styled.div`
  background: ${lightGrey};
  font-family: 'Source Code Pro', monospace;
  font-size: ${polished.rem(20)};
  height: ${polished.rem(500)};
  max-height: ${polished.rem(500)};
  overflow: auto;
  ${column};
  * > textarea:focus {
    outline: none;
  }
`;

const StyledPreview = styled(LivePreview)`
  position: relative;
  padding: 0.5rem;
  background: ${blue};
  color: black;
  height: auto;
  overflow: hidden;
  ${column};
`;

const StyledError = styled(LiveError)`
  display: block;
  padding: ${polished.rem(8)};
  background: ${red};
  color: ${foreground};
  white-space: pre-wrap;
  text-align: left;
  font-size: 0.9em;
  font-family: 'Source Code Pro', monospace;
`;

const scope = { useState }

const code = `
() => {

  return (
    <>
      <strong>Hello!</strong>
      <p>Code a React component on the left to preview it here.</p>
    </>
  )
}
`




const App = () => {
  return (
    <div style={{marginTop: '10%', marginLeft: '2%', marginRight: '2%'}}>
      <StyledProvider code={code} theme={reactLiveHome} scope={scope} language={'jsx'}>
        <LiveWrapper>
          <StyledEditor>
            <LiveEditor />
          </StyledEditor>
          <StyledPreview />
        </LiveWrapper>
        <StyledError />
      </StyledProvider>
    </div>
  )
}

export default App
