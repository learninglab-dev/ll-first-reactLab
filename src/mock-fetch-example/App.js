import React from 'react'
import Layout from './Layout'
import FetchData from './FetchData'


export default function App() {
  return (
    <Layout>
      <FetchData />
    </Layout>
  )
}
