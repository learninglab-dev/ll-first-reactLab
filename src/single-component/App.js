import React, { useState } from 'react'
import { Container } from 'shards-react'


function App() {
  const urls = {
    img: 'https://images.pexels.com/photos/416160/pexels-photo-416160.jpeg',
    gif: 'https://media.giphy.com/media/HZUHce5TyYixq/giphy.gif',
  }
  const [showImg, setShowImg] = useState(true)

  if (showImg) {
    return (
      <Container style={{display: 'flex', padding: '10%', justifyContent: 'space-between', flexWrap: 'wrap', backgroundColor: '#e2e8ed', minHeight: '100vh', margin: 0, flexDirection: 'row', alignItems: 'flex-start', alignContent: 'flex-start' }}>
        <div style={{maxHeight: '500px'}}>
          <button style={{border: '5px solid white', padding: 0, marginBottom: '100px'}} onClick={() => setShowImg(false)}>
            <img src={urls.img} alt='my img' width='300px' />
          </button>
        </div>
      </Container>
    )
  } else {
    return (
      <Container style={{display: 'flex', padding: '10%', justifyContent: 'space-between', flexWrap: 'wrap', backgroundColor: '#e2e8ed', minHeight: '100vh', margin: 0, flexDirection: 'row', alignItems: 'flex-start', alignContent: 'flex-start' }}>
        <div style={{maxHeight: '500px'}}>
          <button style={{border: '5px solid white', padding: 0, marginBottom: '100px'}} onClick={() => setShowImg(true)}>
            <img src={urls.gif} alt='my img' width='300px' />
          </button>
        </div>
      </Container>
    )
  }
}

export default App
