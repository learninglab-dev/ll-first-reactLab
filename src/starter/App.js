import React from 'react'
import Layout from './Layout'
import imgSources from './data'

//the top-level component for our app. I've added some basic styling and a flexbox layout for you so your gifs make a nice grid :)
export default function App() {
  return (
    <Layout>
    </Layout>
  )
}
