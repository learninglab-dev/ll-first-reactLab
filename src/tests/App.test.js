import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import App from '../02_starter/App'

test('it renders and images display', () => {
  const { asFragment, getByAltText } = render(
  <App />
  )
  expect(asFragment()).toMatchSnapshot()
  expect(getByAltText('cute kitten')).toHaveAttribute('src', 'https://images.pexels.com/photos/416160/pexels-photo-416160.jpeg')
  expect(getByAltText('also cute')).toHaveAttribute('src', 'https://images.pexels.com/photos/127027/pexels-photo-127027.jpeg')
})
