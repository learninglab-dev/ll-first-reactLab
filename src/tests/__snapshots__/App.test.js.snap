// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and images display 1`] = `
<DocumentFragment>
  <div
    class="container"
    style="display: flex; padding: 10%; justify-content: space-between; flex-wrap: wrap; background-color: rgb(226, 232, 237); min-height: 100vh; margin: 0px; flex-direction: row; align-items: flex-start; align-content: flex-start;"
  >
    <img
      alt="cute kitten"
      src="https://images.pexels.com/photos/416160/pexels-photo-416160.jpeg"
      width="300px"
    />
    <img
      alt="also cute"
      src="https://images.pexels.com/photos/127027/pexels-photo-127027.jpeg"
      width="300px"
    />
    <img
      alt="not a kitten"
      src="https://images.pexels.com/photos/1564506/pexels-photo-1564506.jpeg"
      width="300px"
    />
  </div>
</DocumentFragment>
`;
